mod-tool/src/main/scala/at/forsyte/apalache/tla/Tool.scala

```diff
@@ -122,36 +110,35 @@ object Tool extends LazyLogging {
             ExitCodes.ERROR
           }
           case Right(()) => {
-            val startTime = LocalDateTime.now()
-            try {
-              cmd match {
-                case parse: ParseCmd =>
-                  runForModule(runParse, new ParserModule, parse)
-
-                case simulate: SimulateCmd =>
-                  // simulation is just a special case of checking, which has additional parameters passed via SimulateCmd
-                  runForModule(runCheck, new CheckerModule, simulate)
-
-                case check: CheckCmd =>
-                  runForModule(runCheck, new CheckerModule, check)
 
-                case test: TestCmd =>
-                  runForModule(runTest, new CheckerModule, test)
-
-                case typecheck: TypeCheckCmd =>
-                  runForModule(runTypeCheck, new TypeCheckerModule, typecheck)
-
-                case server: ServerCmd =>
-                  runForModule(runServer, new CheckerModule, server)
-
-                case constrain: TranspileCmd =>
-                  runForModule(runConstrain, new ReTLAToVMTModule, constrain)
+            val startTime = LocalDateTime.now()
 
-                case config: ConfigCmd =>
-                  configure(config)
+            try {
```

Comment:
To gauge the improvement to readability of this chance, I recommend comparing the revision side by side, and try first to follow the high-level execution sequence through the nested functions invocations in the original, then try to follow the high-level execution sequence that is presented in the changes.